src/workerd/api/node/filesystem.h

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+#pragma once
```

Comment:
Given that there's a possibility of us implementing the WHATWG file system api on top of this also (https://fs.spec.whatwg.org/) it might make sense to move this into `workerd/api` and to not make any part of this C++ layer Node.js specific in any way.